internal/controllers/clusterextension_controller.go

```diff
@@ -165,6 +166,13 @@ func (r *ClusterExtensionReconciler) reconcile(ctx context.Context, ext *ocv1alp
 		setDeprecationStatusesUnknown(&ext.Status.Conditions, "deprecation checks have not been attempted as installation has failed", ext.GetGeneration())
 		return ctrl.Result{}, err
 	}
+
+	if err := r.validateBundle(bundle); err != nil {
```

Comment:
1. Need tests
2. Need to find a real world example of `"olm.constraint"` usage.